### PR TITLE
Switch to Github Actions CI

### PR DESCRIPTION
**Pull request by ehpor**
This PR moves CI from Azure Pipelines to Github Actions. This allows for tighter integration into the Github ecosystem. This also unifies the code between operating systems and reduces workload later on, since Actions are kept up to date by outsiders.

Todo:
- [x] Fix coverage upload.
- [x] Ensure that documentation artifacts look correct.
- [x] Ensure that documentation upload works.
- [x] Fix documentation version number.
- [ ] Change required Github checks.

### CODE REVIEW

**Comment by ehpor** [comment]
@ivalaginja I'll change out the Github required checks after your review.

**Review by raphaelpclt — APPROVED** [review]
I am good with that. GitHub actions logs look ok.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        filter: tree:0
+    - name: Set up conda environment
+      uses: conda-incubator/setup-miniconda@v3
+      with:
+        activate-environment: ci-env
+        python-version: ${{ matrix.python_version }}
+        environment-file: .github/conda_environments/docs_env.yml
```
Comment (ivalaginja) [inline .github/workflows/testing.yml:32]: @ehpor  Is this one supposed to use testing_env instead?

Comment (ehpor) [reply .github/workflows/testing.yml:32]: It should. Fixed in next commit.

**Review by ivalaginja — APPROVED** [review]

**Comment by ehpor** [comment]
Thanks! Changing the required checks now.

**Pull request merged**